in-viewport.js

```diff
@@ -136,49 +136,32 @@ function createInViewport(container) {
     }
 
     var eltRect = elt.getBoundingClientRect();
-    var containerRect = container.getBoundingClientRect();
-
-    var pos = {
-      left: eltRect.left,
-      top: eltRect.top
-    };
-
-    var viewport = {
-      width: offset,
-      height: offset
-    };
+    var viewport = {};
 
     if (container === global.document.body) {
-      viewport.width += global.document.documentElement.clientWidth;
-      viewport.height += global.document.documentElement.clientHeight;
-
-      // We update body rect computing because
-      // when you have relative/absolute childs, you get bad compute
-      // we need to create a new Object, because it's read only
-      containerRect = {
-        bottom: container.scrollHeight,
-        top: 0,
-        left: 0,
-        right: container.scrollWidth
+      viewport = {
+        top: -offset,
+        left: -offset,
+        right: global.document.documentElement.clientWidth + offset,
+        bottom: global.document.documentElement.clientHeight + offset
       };
     } else {
-      pos.left -= containerRect.left;
-      pos.top -= containerRect.top;
-      viewport.width += container.clientWidth;
-      viewport.height += container.clientHeight;
+      var containerRect = container.getBoundingClientRect();
+      viewport = {
+        top: containerRect.top - offset,
+        left: containerRect.left - offset,
+        right: containerRect.right + offset,
+        bottom: containerRect.bottom + offset
```

Comment:
This is the actual bugfix, I know it can be hard to focus on old projects (you want to rewrite a lot, that's normal).

But maybe try next time to only fix this, in a single PR, and then submit the other bits.

Thanks a lot still, that's awesome